src/infrastructure/BinanceClient.go

```diff
@@ -0,0 +1,54 @@
+package infrastructure
+
+import (
+	"btcRate/domain"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+type BinanceClient struct {
+	client  *http.Client
+	baseURL *url.URL
+}
+
+func NewBinanceClient() *BinanceClient {
+	baseUrl := &url.URL{Scheme: "https", Host: "api.binance.com", Path: "/api/v3"}
+
+	binanceClient := &BinanceClient{
+		client:  http.DefaultClient,
+		baseURL: baseUrl,
+	}
+
+	return binanceClient
+}
+
+func (binanceClient *BinanceClient) GetRate(currency string, coin string) (float64, time.Time, error) {
+	path := fmt.Sprintf("/ticker/price?symbol=%s%s", coin, currency)
+	url := binanceClient.baseURL.String() + path
+
+	resp, err := binanceClient.client.Get(url)
+	if err != nil || resp.StatusCode != http.StatusOK {
+		return 0.0, time.Time{}, &domain.EndpointInaccessibleError{Message: "Couldn't access the Binance endpoint"}
+	}
+
+	timestamp := time.Now()
+
+	respBody, _ := io.ReadAll(resp.Body)
```

Comment:
```suggestion
	respBody, err := io.ReadAll(resp.Body)
	if err != nil {
	// ...
	}
```
В деяких місцях відсутня обробка помилок, тобто якщо не вдасться прочитати resp.Body програма продовжить виконання так як наче нічого не сталось

Reply:
Пофксив усі місця в коді